File: scripts/Is Subsequence/Is Subsequence.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isSubsequence = function(s, t) {
    for (let i = 0, n = s.length; i < n; i++)
        if (t.includes(s[i]))
            t = t.slice(t.indexOf(s[i]) + 1);
        else return false;
    
    return true;
}

File: scripts/algorithms/Is Subsequence/Is Subsequence.java
class Solution 
{
    public boolean isSubsequence(String s, String t) 
    {
        int i,x,p=-1;
        if(s.length()>t.length())
            return false;
        for(i=0;i<s.length();i++)
        {
            x=t.indexOf(s.charAt(i),p+1);
            if(x>p)
                p=x;
            else
                return false;
        }
        return true;
    }
}
